### DISCUSSION TITLE
Model-viewer tries to access properties of ThreeRenderer even if it is null

### DISCUSSION THREAD

**Discussion by nicolas-daures** [google/model-viewer#4325] [Q&A]
### Description

Model-viewer tries to access properties of `ThreeRenderer` even if it is `null`.

**Scenario:**
I use Playwright to run automatic tests. When Playwright runs tests on Webkit in headless mode, it cannot instantiate WebGL Context. The `ThreeRenderer` handled by model-viewer is therefore `null`. As its properties are accessed without testing that it is correctly instantiated, uncaught exceptions are raised.

**Issue:**
Since the `Renderer` is a singleton created from the import of model-viewer, an error is thrown and blocks the loading of third parties of the project (in my case, it blocks the automatic tests).

### Version
- model-viewer: v3.1.1

### Browser Affected
- [x] Safari (webkit)

**Comment by elalish** [comment]
Well, I would tend to say if a WebGL context can't be created, then MV has failed, so throwing an error feels like the right thing for a test to do. Certainly I wouldn't trust the tests if the renderer is `null`. 

We also use Playwright for our CI and run on Webkit and it's working fine. Perhaps you can look at our [setup](https://github.com/google/model-viewer/blob/master/packages/model-viewer/web-test-ci.config.mjs#L34).

**Reply by nicolas-daures** [reply]
Thank you for your reply.

The inability to create a WebGL Context came from Playwright's docker image. By updating it, the WebGL context is well created. It's no longer blocking me.

On the other hand, I have a doubt about throwing an error when importing model-viewer because it is difficult to handle the error. Moreover, it also means that we create a WebGL Context even if we don't need it yet.
